aea/registries/base.py

```diff
@@ -117,6 +119,127 @@ def teardown(self) -> None:
         """
 
 
+class ContractRegistry(Registry[PublicId, Contract]):
```

Comment:
This looks like very similar to 
```class ProtocolRegistry(Registry[PublicId, Protocol]):```

I wonder if we could define a class `DefaultRegistry`to inherit from, with common operations already defined so to avoid code duplication. Just a suggestion, not a priority.

Reply:
Agreed, won't do it now though.